pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -502,5 +514,23 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 		}
 	}
 
-	return err
+	return nil
+}
+
+func (s *Reconciler) getCustomUserData() (string, error) {
+	userData := ""
+	if s.scope.MachineConfig.UserDataSecret != nil {
+		var userDataSecret apicorev1.Secret
+
+		if err := s.scope.CoreClient.Get(context.Background(), client.ObjectKey{Namespace: s.scope.Namespace(), Name: s.scope.MachineConfig.UserDataSecret.Name}, &userDataSecret); err != nil {
+			return "", errors.Wrapf(err, "error getting user data secret %s in namespace %s", s.scope.MachineConfig.UserDataSecret.Name, s.scope.Namespace())
+		}
+		if data, exists := userDataSecret.Data["userData"]; exists {
+			userData = string(data)
+		} else {
+			klog.Warningf("Secret %v/%v does not have userData field set. Thus, no user data applied when creating an instance.", s.scope.Namespace(), s.scope.MachineConfig.UserDataSecret.Name)
```

Comment:
I know this is coming from aws but I think we should probably just fail here

Reply:
I thought the reason why it wouldnt want to fail is to support not running bootstrap ignition.  But i cant think of a case of that now. Returning error and fixing tests